packages/cli/src/cmds/preview/preview.ts

```diff
@@ -6,147 +6,495 @@
+      constants.authStartSuccessMessage,
+      true
+    );
+
+    const backendStartBar = DialogManagerInstance.createProgressBar(constants.backendStartTitle, 1);
+    const backendStartStartCb = commonUtils.createTaskStartCb(
+      backendStartBar,
+      constants.backendStartStartMessage
+    );
+    const backendStartStopCb = commonUtils.createTaskStopCb(
+      constants.backendStartTitle,
+      backendStartBar,
+      constants.backendStartSuccessMessage,
+      true
+    );
+
+    const botStartBar = DialogManagerInstance.createProgressBar(constants.botStartTitle, 1);
+    const botStartStartCb = commonUtils.createTaskStartCb(
+      botStartBar,
+      constants.botStartStartMessage
+    );
+    const botStartStopCb = commonUtils.createTaskStopCb(
+      constants.botStartTitle,
+      botStartBar,
+      constants.botStartSuccessMessage,
+      true
+    );
+
+    const results = await Promise.all([
+      frontendStartTask?.waitFor(
+        constants.frontendStartPattern,
+        frontendStartStartCb,
+        frontendStartStopCb
+      ),
+      authStartTask?.waitFor(constants.authStartPattern, authStartStartCb, authStartStopCb),
+      backendStartTask?.waitFor(
+        constants.backendStartPattern,
+        backendStartStartCb,
+        backendStartStopCb
+      ),
+      await botStartTask?.waitFor(constants.botStartPattern, botStartStartCb, botStartStopCb),
+    ]);
+    const fxErrors: FxError[] = [];
+    for (const result of results) {
+      if (result?.isErr()) {
+        fxErrors.push(result.error);
+      }
+    }
+    if (fxErrors.length > 0) {
+      return err(errors.PreviewCommandFailed(fxErrors));
+    }
+    return ok(null);
+  }
+
+  private async openTeamsWebClient(
+    tenantIdFromConfig: string | undefined,
+    teamsAppId: string
+  ): Promise<Result<null, FxError>> {
+    let sideloadingUrl = constants.sideloadingUrl.replace(
+      constants.teamsAppIdPlaceholder,
+      teamsAppId
+    );
+
+    let tenantId, loginHint: string | undefined;
+    try {
+      const tokenObject = (await AppStudioTokenInstance.getStatus())?.accountInfo;
+      if (tokenObject) {
+        // user signed in
+        tenantId = tokenObject.tid as string;
+        loginHint = tokenObject.upn as string;
+      } else {
+        // no signed user
+        tenantId = tenantIdFromConfig;
+        loginHint = "login_your_m365_account"; // a workaround that user has the chance to login
+      }
+    } catch {
+      // ignore error
+    }
+
+    if (tenantId && loginHint) {
+      sideloadingUrl = sideloadingUrl.replace(
+        constants.accountHintPlaceholder,
+        `appTenantId=${tenantId}&login_hint=${loginHint}`
+      );
+    } else {
+      sideloadingUrl = sideloadingUrl.replace(constants.accountHintPlaceholder, "");
+    }
+
+    const sideloadingBar = DialogManagerInstance.createProgressBar(constants.sideloadingTitle, 1);
+    await sideloadingBar.start(`${constants.sideloadingStartMessage}${sideloadingUrl}`);
```

Comment:
Anyway to highlight the url and make it clickable? Like https://github.com/OfficeDev/TeamsFx/pull/1381

Reply:
Updated.
![image](https://user-images.githubusercontent.com/37978464/123598585-43ef1c00-d827-11eb-8a25-96b15de07cae.png)